### DISCUSSION TITLE
Watchdog / Auto Fail for "Execute a process"

### DISCUSSION THREAD

**Discussion by usbrandon** [apache/hop#4747] [Ideas]
Executing a process is a very useful way of extending Hop without waiting for someone to code a solution that works for whatever the us use case there is.  However, one of the problems I have encountered is if a process runs long or hangs, Hop also hangs because it is waiting around on the process to finish.  I wish there were an elegant way to have an executed process kill itself after a time limit.

**Comment by hansva** [comment]
Great idea! Feel free to create a feature request for this!

**Reply by usbrandon** [reply]
Added: https://github.com/apache/hop/issues/4748
